### ISSUE TITLE
Is it possible to get notified or automatically update modules when updates are released?

### ISSUE THREAD

**Issue by asdsadsaasd12312132** [topjohnwu/Magisk#3656]
I noticed that several of my modules were outdated by a year because I hadn't checked in so long. I would like to keep them updated but don't want to have to check manually frequently. Is it possible to get notifications or automatically update?

**Comment by Haus3master** [comment]
I hope @topjohnwu labels this query as 'enhancement' and hopefully brings this feature in the next update....

**Comment by IanWorthington** [comment]
Given the current problems with 21.2, I would suggest that letting it update automatically may not be a good idea :(

**Comment by vvb2060** [comment]
Before updating, you should read the changelog, automatic updates are not a good idea.

**Comment by IanWorthington** [comment]
> Before updating, you should read the changelog, automatic updates are not a good idea.

As I did for 21.2 last night.  Looked innocuous.  How wrong can one be? :(

Better advice might be wait until you have a problem and then see what's the latest version which hasn't got issues raised against it either here or on xda?  To that end, the ability in the manager to install versions other than the most recent might be useful.

**Comment by selurvedu** [comment]
@vvb2060 I'd like to be notified that an update is available. Deciding whether to install it or not should be up to the user.

**Comment by osm0sis** [comment]
Well, there is no modules repo within the app anymore, but there is a plan for adding 3rd party repos as a setting within the app, so I guess this can remain open for that.

**Comment by yujincheng08** [comment]
https://github.com/topjohnwu/Magisk/commit/bc0c1980db62c15c8dd55ffc7f53521410f5706c

**Comment by osm0sis** [comment]
Hmm, not sure if that actually notifies when there's an update or not from reading through the commit. 🤔

**Comment by zkollz** [comment]
> I noticed that several of my modules were outdated by a year because I hadn't checked in so long. I would like to keep them updated but don't want to have to check manually frequently. Is it possible to get notifications or automatically update?
> 

how about you try the fox magisk module manager?

**Comment by MartinX3** [comment]
Even if there is no repo anymore and modules contain their own update url, there should be a setting in the App to as example check every 24 hours for new module versions and notify the user.

**Comment by breversa** [comment]
There's a app for this: _Fox's Magisk Module Manager_ 
https://f-droid.org/packages/com.fox2code.mmm.fdroid/

**Comment by osm0sis** [comment]
Again, modules can still update themselves in the current format, so a feature for a notification when an update is available is still a valid request.

**Comment by MartinX3** [comment]
Also they don't autoupdate (at my side).
In my case I need to regular visit the Magisk App and wait in the Modules Section that every module got checked for a new version.

**Comment by osm0sis** [comment]
> Also they don't autoupdate (at my side). In my case I need to regular visit the Magisk App and wait in the Modules Section that every module got checked for a new version.

Exactly, so the current request would be for that update check to occur in the background and for a notification when an update is available.

**Comment by xerta555** [comment]
> > Also they don't autoupdate (at my side). In my case I need to regular visit the Magisk App and wait in the Modules Section that every module got checked for a new version.
> 
> Exactly, so the current request would be for that update check to occur in the background and for a notification when an update is available.

If you look at it from the outside, that users are now checking for updates themselves or using another application or not checking at all, you realize that there is a problem.

**Comment by DemonoidMaster** [comment]
Old thread i know but i agree it would be nice if it could check for module updates at the same time as  for magisk updates through a built in timer the user can set, why isn't there a setting for that ? Magisk just checks for updates of itself whenever it wants but how often, dunno, doesn't tell the user, would be nice having it check weekly/monthly instead of everyday if that's what it does

**Comment by Gouster4** [comment]
There sould be options to:
1. Notify for updates and let user decide
2. Autoupdates (can come with warning that it can break things).  And notification about modules was updated and requires reboot.
SafeMode boot can be used to recover from bad update. I personally would rather use safe mode to fix something then manyally update modules. So option for that can be usefull.
